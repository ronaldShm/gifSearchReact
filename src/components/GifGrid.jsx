import { GiftItem } from "./GiftItem"
import { useFetchGifs } from "../hooks/useFetchGifs"

export const GifGrid = ({ category }) => {
    const {images, isLoading} = useFetchGifs(category)
    console.log({isLoading})

    
    
    return (
        <>
            <h3>{ category }</h3>
            {
                isLoading
                ? (<h2>Cargando...</h2>)
                : null
            }

            <div className="card-grid">
                {
                    images.map( (image) => (
                        <GiftItem 
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>
        </>
    )
}
