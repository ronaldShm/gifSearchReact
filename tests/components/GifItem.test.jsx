import { render, screen } from '@testing-library/react'
import { GiftItem } from '../../src/components'

describe('Pruebas en <GifItem/>', () => {
    
    const title = 'Dragon Ball Z'
    const url = 'https://www.google.com/'
    
    test('Debe de hacer match con el snapshot', () => { 
        const {container} = render( <GiftItem title={title} url={url}/>)
        expect(container).toMatchSnapshot()
    })

    test('Debe de mostrar la imagen con el URL y ALT indicado', () => { 
        render( <GiftItem title={title} url={url}/>)
        // screen.debug()
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(alt)
        
    })

    test('Debe de mostrar el titulo en el componente', () => { 
        render( <GiftItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();

    })
 })