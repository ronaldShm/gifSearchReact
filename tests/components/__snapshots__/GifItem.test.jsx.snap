// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball Z"
      src="https://www.google.com/"
    />
    <p>
      Dragon Ball Z
    </p>
  </div>
</div>
`;

exports[`Pruebas en GifItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball Z"
      src="https://www.google.com/"
    />
    <p>
      Dragon Ball Z
    </p>
  </div>
</div>
`;
